prog/test/ha_postgres_resource.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::Test::HaPostgresResource < Prog::Test::Base
+  semaphore :destroy
+
+  def self.assemble
+    postgres_test_project = Project.create(name: "Postgres-Test-Project")
+    postgres_service_project = Project[Config.postgres_service_project_id] ||
+      Project.create(name: "Postgres-Service-Project") { _1.id = Config.postgres_service_project_id }
+    minio_service_project = Project[Config.minio_service_project_id] ||
+      Project.create(name: "MinioServiceProject") { _1.id = Config.minio_service_project_id }
+
+    frame = {
+      "postgres_service_project_id" => postgres_service_project.id,
+      "postgres_test_project_id" => postgres_test_project.id,
+      "minio_service_project_id" => minio_service_project.id
+    }
+
+    Strand.create_with_id(
+      prog: "Test::HaPostgresResource",
+      label: "start",
+      stack: [frame]
+    )
+  end
+
+  label def start
+    st = Prog::Minio::MinioClusterNexus.assemble(postgres_service_project.id,
+      "minio-0", "hetzner-fsn1", "admin", 256, 1, 1, 1, "standard-2")
+
+    update_stack({"minio_cluster_id" => st.id})
+    hop_wait_minio_cluster
+  end
+
+  label def wait_minio_cluster
+    if minio_cluster.strand.label == "wait"
+      hop_create_postgres_resource
+    else
+      nap 10
+    end
+  end
+
+  label def create_postgres_resource
+    st = Prog::Postgres::PostgresResourceNexus.assemble(
+      project_id: frame["postgres_test_project_id"],
+      location: "hetzner-fsn1",
+      name: "postgres-test-ha",
+      target_vm_size: "standard-2",
+      target_storage_size_gib: 128,
+      ha_type: "async"
+    )
+
+    update_stack({"postgres_resource_id" => st.id})
+    hop_wait_postgres_resource
+  end
+
+  label def wait_postgres_resource
+    num_servers = postgres_resource.servers.count
+    required_servers = 1 + postgres_resource.required_standby_count
+    nap 10 if num_servers != required_servers || postgres_resource.servers.filter { _1.strand.label != "wait" }.any?
```

Comment:
```
postgres_resource.servers.all?{ _1.strand.label == "wait" }
```

Reply:
I had this earlier but it failed intermittently because the standby is created in the `wait` label, so the test would proceed to trigger failover before standby was ready

Reply:
Makes sense. As a small nitpick, I think `server_count` and `required_server_count` would be better variable names.